### PR TITLE
[bug] binance errors

### PR DESCRIPTION
**Pull request by danielisaacgeslin**
https://github.com/user-attachments/assets/f575ef4a-8444-4986-877a-0736f5e2d47b

### CODE REVIEW

File: wormhole-connect/src/hooks/useIsMounted.ts
Comment (emreboga) [inline wormhole-connect/src/hooks/useIsMounted.ts:1]: @danielisaacgeslin Thanks for separating this hook out, I believe we can use it in others as well 👍

File: wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import AlertBanner from 'components/v2/AlertBanner';
+import { Stack, Typography, useTheme } from '@mui/material';
+import { toFixedDecimals } from 'utils/balance';
+
+export type WalletBalanceWarningProps = {
+  isCheckingBalance: boolean;
+  isBalanceEnough: boolean;
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx]: Nitpick: Would you consider a different naming, `hasSufficientBalance`?

File: wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  walletBalance?: number;
+  networkCost?: number;
+  symbol?: string;
+};
+
+export default function WalletBalanceWarning({
+  isCheckingBalance,
+  isBalanceEnough,
+  walletBalance,
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx:10]: Nitpick: Is this the token's balance in the wallet? In that case do you it'd be better to name it `tokenBalance`?

File: wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  networkCost,
+  symbol,
+}: WalletBalanceWarningProps) {
+  const theme = useTheme();
+  const content = isCheckingBalance
+    ? `Checking if wallet balance is enough...`
+    : `Your wallet balance is not enough to cover the network cost. Please add more funds to your wallet.`;
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx]: "Insufficient token balance to cover the network cost. Please add more funds to your wallet." ?

File: wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+export default function WalletBalanceWarning({
+  isCheckingBalance,
+  isBalanceEnough,
+  walletBalance,
+  networkCost,
+  symbol,
+}: WalletBalanceWarningProps) {
+  const theme = useTheme();
+  const content = isCheckingBalance
+    ? `Checking if wallet balance is enough...`
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx]: "Checking token balance for network costs." ?

File: wormhole-connect/src/hooks/useBalanceChecker.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import config from 'config';
+import { useIsMounted } from './useIsMounted';
+
+export const useBalanceChecker = (
+  quote?: routes.Quote<
+    routes.Options,
+    routes.ValidatedTransferParams<routes.Options>
+  >,
+): {
+  feeSymbol?: string;
```
Comment (emreboga) [inline wormhole-connect/src/hooks/useBalanceChecker.ts:13]: Aren't these return props very similar (if not the same) to `WalletBalanceWarningProps` below? Can you import it here as well? Or maybe move it to `wormhole-connect/src/config/types.ts`?

File: wormhole-connect/src/hooks/useBalanceChecker.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      Object.values(config.tokens),
+    )?.symbol;
+  }, [quote]);
+
+  useEffect(() => {
+    setCheckingBalance(true);
+    (async () => {
+      try {
+        const wallet = getWalletConnection(TransferWallet.SENDING);
+        if (!wallet || !quote?.relayFee?.amount) return;
```
Comment (emreboga) [inline wormhole-connect/src/hooks/useBalanceChecker.ts]: You can check and return for `!quote?.relayFee?.amount` even before getting the wallet.

File: wormhole-connect/src/hooks/useBalanceChecker.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+        const cost = amount.whole(quote.relayFee.amount);
+        const balance = parseFloat(await wallet.getBalance());
+
+        if (isMounted.current) setState({ balance, cost });
+      } catch (e) {
+        console.error(e);
+        if (isMounted.current) setState(null);
+      } finally {
+        if (isMounted.current) setCheckingBalance(false);
```
Comment (emreboga) [inline wormhole-connect/src/hooks/useBalanceChecker.ts:51]: Is there a way to test this case? Like getting the isMounted hook unmounted while isCheckingBalance is true? It's a very very edge case I know, but do you think would it be possible to unmount the hook without the component still being mounted? In that case it might get stuck in checking state.

Comment (danielisaacgeslin) [reply wormhole-connect/src/hooks/useBalanceChecker.ts:51]: it wouldn't be possible to unmount the hook while the compo is mounted in a real case. Hooks can't be conditional and they're bound together in the same render branch therefore hooks would only unmount along with the compo that called them.

That being said, a realistic way to test this (nvm fast fingers) would be to set an arbitrary long wait in try clause whilst unmounting through the UI (pressing the back button i think)

File: wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx (hunk 2 above)
Comment (danielisaacgeslin) [reply wormhole-connect/src/views/v2/Bridge/ReviewTransaction/WalletBalanceWarning.tsx:10]: leaving this one as is as per slack comments

**Review by emreboga — APPROVED** [review]
